test/files/rules/noinferrabletypes.test.ts

```diff
@@ -0,0 +1,21 @@
+// errors, inferrable type is declared
+let x: number = 7;
+let y: boolean = false;
+let z: string = "foo";
+
+// errors, types are inferrable
+function foo (a: number = 5, b: boolean = true, c: string = "bah") { }
+
+// not errors, inferrable type is not declared
+let _x = 7;
+let _y = false;
+let _z = "foo";
+
+// not error, type is not inferrable
+let weird: any = 123;
+
+// not errors, inferrable type is not declared
+function bar(a = 5, b = true, c = "bah") { }
+
+// not errors, types are not inferrable
+function baz(a: any = 5, b: any = true, c: any = "bah") { }
```

Comment:
I thought about this as well. We could probably add a feature for functions w/ one return statement where the declared return type is the same as the type of the return statement. Or reduce the scope and only do it for void, string, number, and boolean.